js/app.js

```diff
@@ -0,0 +1,429 @@
+                $obj[0].innerHTML = $obj.attr("old_value");  
+            } else if (self.validItem(val)) {
+                $obj[0].innerHTML = $obj[0].innerHTML.toUpperCase()   
+                myList = updateArray();
+                updateTotal();
+            } else {
+                $obj[0].innerHTML = $obj.attr("old_value");   
+                myList = updateArray();
+                updateTotal();
+            }
+        } else if ($obj.hasClass("price")) {
+            const price = Number(val.replace(/[^0-9\.-]+/g,"")),
+                  $coupon = $obj.parent().find(".coupon_active"),
+                  $total = $obj.parent().find(".total");
+            let discount = 0;
+
+            if ($coupon[0] !== undefined) {
+                  discount = Number($coupon[0].innerHTML.replace(/[^0-9\.-]+/g,""))
+                }
+            
+            if (self.validPrice(Number(val.replace(/[^0-9\.-]+/g,""))) === false) {
+                $obj[0].innerHTML = $obj.attr("old_value");   
+            } else {
+                $total[0].innerHTML = "$" + (price * (1 - (discount / 100))).toFixed(2);
+                $obj[0].innerHTML = "$" + Number(val.replace(/[^0-9\.-]+/g,"")).toFixed(2); 
+            }
+            myList = updateArray();
+            updateTotal();
+        } else if ($obj.hasClass("coupon") || $obj.hasClass("coupon_active")) {
+            
+            const discount = Number(val.replace(/[^0-9\.-]+/g,"")),
+                  $price = $obj.parent().find(".price"),
+                  price = Number($price[0].innerHTML.replace(/[^0-9\.-]+/g,"")),
+                  $total = $obj.parent().find(".total");
+            
+            if (self.validCoupon(discount) === false) {
+                $obj[0].innerHTML = $obj.attr("old_value");   
+            } else if (discount > 0) {
+                $obj[0].innerHTML = discount + "%";
+                $total[0].innerHTML = "$" + (price * (1 - (discount / 100))).toFixed(2);
+                $obj[0].setAttribute("class", "coupon_active");
+            } else {
+                $obj[0].innerHTML = "";
+                $total[0].innerHTML = "$" + price.toFixed(2);
+                $obj[0].setAttribute("class", "coupon");
+            }  
+            myList = updateArray();
+            updateTotal();
+            
+        } 
+    }
+    
+}
+
+    const validBudget = (budget) => {
+        let errText = "";
+        if (typeof budget != "number") {
+            errText = "argument 'budget' != number|budget:"+budget;
+        } else if (budget < 0) {
+            errText = "budget must be >= 0|budget:"+budget;
+        } else if (Math.round(budget * 100) / 100 != budget) {
+            errText = "budget may not exceed hundredths|budget:"+budget;
+        }
+
+        if (errText.length != 0) {
+            alert(errText);
+            //throw error;
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    $(document).ready(function() {

```

Comment:
Hm, adding multiple `$(document).ready()` functions isn't usually a good idea. While they stack and will work, it's a best practice to consolidate such functionality into one function. If you want to break them out so the ready function isn't a monolith, you should have one ready function that calls other functions in turn.